src/modules/i18n/loadLocales.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+
+import { reduce, forEach, merge } from 'lodash'
+import defaultLocales from '../../../locales/locales.json'
+import buildConfig from '../app/constants/buildConfig'
+import type { LocalesType } from '../../../build-configs/configs/BuildConfigType'
+
+type TransformedLocalesType = { [language: string]: { [namespace: string]: { [key: string]: string } } }
+
+/**
+ * Transform locale resources from our internal locales format to be i18next compatible.
+ * @param {object} locales in our format: namespace -> languageCode -> key -> value
+ * @returns {object} transformed locales in the format: languageCode -> namespace -> key -> value
+ */
+const transformLocales = (locales: LocalesType): TransformedLocalesType => reduce(
+  locales,
+  (transformedLocales, namespace, namespaceName) => {
+    forEach(namespace, (language, languageCode) => {
+      transformedLocales[languageCode] = {
+        ...transformedLocales[languageCode],
+        [namespaceName]: language
+      }
+    })
+    return transformedLocales
+  },
+  {}
+)
+
+const loadLocales = (): TransformedLocalesType => {
+  const localesOverride = buildConfig().localesOverride
+  const locales = localesOverride ? merge(defaultLocales, localesOverride) : defaultLocales
```

Comment:
Merge ignores values in localesOverrides if they are missing in defualtLocales: https://lodash.com/docs/4.17.15#merge

We could throw a warning here if localesOverride contains values which do not exist in defaultLocales.

Reply:
Do you know whether there is a lodash function to do this?

Reply:
AFAICS there is none. Do you want me to check recursively? Or should I just add a comment?

Reply:
I think a comment should be enough.